entity-server-api/src/main/java/org/terracotta/entity/PassiveServerEntity.java

```diff
@@ -36,9 +36,15 @@
    *  under a given key, the messages it creates and sends for synchronization will be executed in that key, on the passive,
    *  via this invoke() entry-point.</p>
    *
+   * @param clientDescriptor source instance from which the invocation originates.
+   * @param currentOrderedId current id this invoke is part of
+   * @param eldestOrderedId earliest active id for this client
    * @param message The message from a client or upstream active server
    */
-  void invoke(M message) throws EntityUserException;
+  void invoke(ClientDescriptor clientDescriptor,
```

Comment:
@albinsuresh. If I understand the question correctly, it's up to the entity to create and maintain the association with the message it creates for ientitymessenger.